app/screens/channel/channel_nav_bar/channel_drawer_button/channel_drawer_button.js

```diff
@@ -12,67 +11,54 @@ import {
 import Icon from 'react-native-vector-icons/Ionicons';
 
 import Badge from 'app/components/badge';
-import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import PushNotifications from 'app/push_notifications';
 import {preventDoubleTap} from 'app/utils/tap';
 import {makeStyleSheetFromTheme} from 'app/utils/theme';
 
 import telemetry from 'app/telemetry';
 
-import {getUnreadsInCurrentTeam} from 'mattermost-redux/selectors/entities/channels';
-import {getCurrentTeamId, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
-
-class ChannelDrawerButton extends PureComponent {
+export default class ChannelDrawerButton extends PureComponent {
     static propTypes = {
-        currentTeamId: PropTypes.string.isRequired,
         openDrawer: PropTypes.func.isRequired,
-        messageCount: PropTypes.number,
-        mentionCount: PropTypes.number,
-        myTeamMembers: PropTypes.object,
+        badgeCount: PropTypes.number,
         theme: PropTypes.object,
         visible: PropTypes.bool,
     };
 
     static defaultProps = {
+        badgeCount: 0,
         currentChannel: {},
         theme: {},
-        messageCount: 0,
-        mentionCount: 0,
         visible: true,
     };
 
+    componentDidMount() {
+        if (this.props.badgeCount > 0) {
```

Comment:
Is this check necessary? Wouldn't we want to set the badge even if the count is 0?

Reply:
that's why we have the if statement in the componentDidMount, we want to avoid this from setting it to 0 if the mention count is still loading to avoid clearing pending notifications in the notification manager